src/routes/api/sections/querySection.ts

```diff
@@ -1,58 +1,113 @@
         });
+
         if (!sectionInformation) {
             res.status(404);
             return;
         }
 
-        if (startsAt && endsAt) {
-            const start = new Date(startsAt as string);
-            const end = new Date(endsAt as string);
-            // Validate date strings
-            if (start === null || end === null) {
-                res.status(400).json({ error: 'validation failed, variable types are incorrect' });
+        const start = DateTime.fromISO(String(startsAt), {zone: 'America/Toronto'});
+        const end = DateTime.fromISO(String(endsAt), {zone: 'America/Toronto'});
+        // Validates start and end dates
+        if (start === null || end === null) {
+            res.status(400).json({ error: 'validation failed, variable types are incorrect' });
+            return;
+        } else {
+            // Retrives room that the section corresponds to
+            const room = await Room.findOne({ _id: sectionInformation.roomId }).catch((error) => {
+                res.status(404).json({ error });
                 return;
-            } else {
-                // Retrieves all time blocks for the room section within the time range given in the input
-                const sectionTimeblocks = await timeblockModel
-                    .find({
+            });
+
+            if(!room) {
+                res.status(404);
+                return;
+            }
+            
+            let currHourStart = start;
+            let currHourEnd = currHourStart;
+            currHourEnd = currHourEnd.plus({ hours: 1});
+
+            const timeblocks = [];
+            
+            // Iterate through the hours in the given time range
+            while (currHourStart < end) {
+                // Finds schedule start and end for the day that the current hour falls on
+                const currWeekDay = currHourStart.weekday;
+                let scheduleDayStart = 0;
+                let scheduleDayEnd = 0;
+                for (const day of room.schedule) {
+                    if (day.dateOfWeek == currWeekDay) {
+                        scheduleDayStart = day.start;
+                        scheduleDayEnd = day.end;
+                    }
+                }
+                
+                // Checks if the hour falls under an open time for the room and if it does adds hour to the list of available times
+                if (!room.closed && (currHourStart.weekday == currHourEnd.weekday) && (currHourStart.hour >= scheduleDayStart && currHourEnd.hour <= scheduleDayEnd)) {
+                    // Check for timeblock in database
+                    const bookedTimeblock = await timeblockModel
+                    .findOne({
                         sectionId: sectionInformation._id,
-                        startsAt: { $gte: start },
-                        endsAt: { $lte: end },
+                        startsAt: currHourStart.toJSDate(),
+                        endsAt: currHourEnd.toJSDate()
```

Comment:
This isn't very efficient for querying and will introduce a lot of latency. We should find an array of booked timeblocks outside of the loop between the specified dates (you can use `$gte` and `$lte` to query) and then search through that array.